openlibrary/plugins/worksearch/schemes/works.py

```diff
@@ -489,7 +494,7 @@ def convert_work_query_to_edition_query(work_query: str) -> str:
             )
             new_params.append(('editions.rows', '1'))
             new_params.append(('editions.fl', ','.join(edition_fields)))
-
+        remove_work_prefix_from_query(new_params, self.field_name_map)
```

Comment:
Ah this isn't quite the right flow. Somewhere up above, it says:

```py
new_params.append(('workQuery', str(work_q_tree)))
```

_That's_ the query we want to modify. Something like:

```py
def remove_work_prefix(field: str) -> str:
    return field.partition('.')[2] if field.startswith('work.') else field

new_params.append(('workQuery', str(luqum_replace_field(deepcopy(work_q_tree), remove_work_prefix))))
```

Where `luqum_replace_field` is a new function traverses the parse tree and uses the provided function to change the field name. That can live in `query_utils.py` ; use `escape_unknown_fields` as a template to see how to traverse the parse tree!